### PR TITLE
Support client credentials in body

### PR DESCRIPTION
**Pull request by alexandre-kivra**

### CODE REVIEW

**Comment by alexandre-kivra** [comment]
> `expires_in` is actually a relative value (https://tools.ietf.org/html/rfc6749#section-5.1) but as far as I can see it's treated as an absolute value. Which would explain why I can't get it to return the cached token :)

That's a very nice catch. I'll look into it tomorrow morning

File: src/oauth2c_token_cache.erl [hunk 1]
```diff
@@ -128,10 +127,10 @@ get_token(Key, ExpiryTimeLowerLimit) ->
       {error, not_found}
   end.
 
-get_expires_in(undefined, DefaultTTL) ->
+get_expire_time(undefined, DefaultTTL) ->
   erlang:system_time(second) + DefaultTTL;
-get_expires_in(ExpiryTime, _DefaultTTL) ->
-  ExpiryTime.
+get_expire_time(ExpireTime, _DefaultTTL) ->
+  ExpireTime.
```
Comment (jakobsvenning) [inline src/oauth2c_token_cache.erl:133]: Isn't ExpireTime a relative value? Shouldn't this function return erlang:system_time(second) + ExpireTime to get an absolute expire time?

Comment (alexandre-kivra) [reply src/oauth2c_token_cache.erl:133]: So I reinstated the ExpireTime as the absolute value when the token expires. The change is that after we do the request to get the token we immediately calculate the ExpireTime and discard the expires_in value. At least that's what I'm trying to do. The code is complicated unfortunately.

Comment (jakobsvenning) [reply src/oauth2c_token_cache.erl:133]: I missed that you are making the time absolute in the `do_retrieve_access_token ` function. Then everything is in order.

**Pull request merged**